File: backend/domain/schemas/product.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date

class ProductTrackRequest(BaseModel):
    url: str

class ProductResponse(BaseModel):
    id: int
    musinsa_id: str
    url: str
    title: Optional[str] = None
    brand: Optional[str] = None
    thumbnail_url: Optional[str] = None
    image_urls: List[str] = []
    original_price: Optional[int] = None
    is_garment_modeled: bool = False
    current_price: Optional[int] = None
    discount_rate: Optional[int] = None

class ProductListResponse(BaseModel):
    products: List[ProductResponse]
    total: int

class PriceLogResponse(BaseModel):
    id: int
    price: int
    discount_rate: Optional[int]
    captured_at: date

class PriceHistoryResponse(BaseModel):
    product_id: int
    title: Optional[str]
    history: List[PriceLogResponse]
    min_price: Optional[int]
    max_price: Optional[int]
    min_date: Optional[str]
    max_date: Optional[str]

File: frontend/lib/types/product.ts
/**
 * Product related types
 */

export interface Product {
    id: number;
    musinsa_id: string;
    url: string;
    title: string | null;
    brand: string | null;
    thumbnail_url: string | null;
    is_garment_modeled: boolean;
    current_price: number | null;
    discount_rate: number | null;
}

export interface PriceLog {
    id: number;
    price: number;
    discount_rate: number | null;
    captured_at: string;
}

export interface ProductListResponse {
    products: Product[];
    total: number;
}

export interface PriceHistoryResponse {
    product_id: number;
    title: string;
    history: PriceLog[];
}
